runtime/vm/visible.c

```diff
@@ -360,7 +378,10 @@ loadAndVerifyNestHost(J9VMThread *vmThread, J9Class *clazz, UDATA options)
 
 		/* If a problem occurred in nest host verification then the nest host value is invalid */
 		if (J9_VISIBILITY_ALLOWED == result) {
-			clazz->nestHost = nestHost;
+			*nestHostFound = nestHost;
+			if (cacheNestHotsInClass) {
```

Comment:
if its not a hidden class, cant we cache it?

Reply:
`cacheNestHotsInClass` is initialized to TRUE. It is only set to FALSE at line 308 where `hiddenNestMate` is true and extensions are enabled. This is the only case that the nest host can points to stale class. For other cases, we can safely cache nest host in `class->nestHost`.

Reply:
ah okay